src/uu/cp/src/cp.rs

```diff
@@ -2549,10 +2549,10 @@ fn handle_no_preserve_mode(options: &Options, org_mode: u32) -> u32 {
             const MODE_RW_UGO: u32 =
                 (S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH) as u32;
             const S_IRWXUGO: u32 = (S_IRWXU | S_IRWXG | S_IRWXO) as u32;
-            if is_explicit_no_preserve_mode {
-                return MODE_RW_UGO;
+            return if is_explicit_no_preserve_mode {
+                MODE_RW_UGO
             } else {
-                return org_mode & S_IRWXUGO;
+                org_mode & S_IRWXUGO
```

Comment:
Actually, this is duplicated code isn't it from 2526 ? Both `cfg`s share the same snippet 🤔

Reply:
Mhh the logic seems actually the same. My changes here were limited to modifying the return expression in the second case to comply with Clippy's suggestions, but the only difference between the two is the `not` on the first `cfg` condition. I don't know if there is a reason for that.

Reply:
Yes, the `as u32` cast is redundant with the variable type declaration.

I think it's OK to remove the duplicated code here, and get rid of the `#[cfg(...)]`.

Reply:
Seems that it's needed because on macOS and FreeBSD they are actually u16 (see https://github.com/gabelluardo/coreutils/actions/runs/16325706038/job/46115495345)